### DISCUSSION TITLE
Does this PG&E gas meter work? Where to put sensor?

### DISCUSSION THREAD

**Discussion by eigenphase** [tronikos/esphome-magnetometer-water-gas-meter#32] [Q&A]
I have this PG&E gas meter:

![IMG_1703](https://github.com/user-attachments/assets/75906136-60a3-45ff-a5d4-c805a17dc18e)

I do not have the sensors yet but I tried to plot the magnetic field with the “Sensors” app on my iPhone. I held it roughly where it’s shown on the README but I don’t see anything specific. For one, I’m not even sure what to look for and then it’s a different meter.

**Comment by tronikos** [comment]
It should work. Place it in the center of the meter. Try both front and back.

If you hold the phone steady on your desk, the magnetic field strength for the 3 axis should be flat lines. If you hold it steady by the meter while the gas is running you should see sine graphs.

**Reply by eigenphase** [reply]
Oh thank you, yes definitely didn’t see anything sinusoid like.

when you say center of the meter, do you mean the top part (with the gauges) or the met bottom body?

and when you say front/back are you referring to the phone or the meter? (Meter is fairly close to the wall but I think I could still fit phone).

Ill try to get a box and lean the phone towards the meter and show the graphs here

**Reply by tronikos** [reply]
Center, front and back, of the bottom body. Keep in mind since you don't really know where the sensor is on your phone you will have to try several different spots. Keep your phone flat on the meter body surface and move it around pausing for a few seconds to observe the sinus graph or flat line. It would also help if you run your gas stove, gas heater or whatever gas appliance you have while you do this.

**Reply by tronikos** [reply]
See also https://community.home-assistant.io/t/can-this-gas-meter-be-made-smart-in-any-way/562397/6 or other similar posts in the forum if you search for: gas magnetometer 

I personally don't have gas but many have successfully used this project for gas meters.

**Reply by eigenphase** [reply]
I tried multiple spots: I assume only the the part that's below the dials ("belly") is of interest. I tried placing the sensor at the very top but there is absolutely no signal.

At the "belly", I tried bottom, middle and top both front side and backside. The strongest signal I get is on the backside in the middle. However, this just gives me 20uT peak-to-peak. This explains why I didn't see anything with my phone. The signal is just so weak. With my water meter I am getting 200uT peak-to-peak, i.e. the signal is 10x smaller in magnitude! Example:

<img width="434" alt="image" src="https://github.com/user-attachments/assets/7a895fe8-5771-4121-af69-52bdb84a4834">

Z-axis max: -6.6uT, min: -27.8uT, ∆=21.2uT peak-to-peak.

It seems with the "Calibrate" button did not even work. Nothing happens when I click it. Does it require a certain magnitude?
Things seem to work though if I manually calculate and enter the thresholds as you described in the README.

But I am really worried that the SNR is so bad. I know you do not have a gas meter but do you know if gas meters generally much much weaker magnetic field or if I am doing anything wrong? Actually this is a steel body. Shouldn't such steel body heavily attenuate magnetic fields?

**Reply by mattdevo1** [reply]
Hey @segdy, did you have any luck with this? I'm currently trying to set this up for exactly the same PG&E gas meter that you have, with limited luck. Would love to know if you got it working reliably, and how. Thanks!

**Reply by eigenphase** [reply]
Yes, I just went ahead with the weak signal. I had 2 weeks in Dec when I didn't have any gas usage (even though I had). Not sure if it was related to this.

**Reply by mattdevo1** [reply]
Did you have to manually calibrate it? I'm having a hard time getting the calibration right. Any chance you could send a photo of where you located the sensor? Sounds like you have it on the middle on the back, but I'm only getting 5-10uT variance peak-to-peak.

**Reply by eigenphase** [reply]
<img width="504" alt="image" src="https://github.com/user-attachments/assets/c2655687-5b03-4899-b183-942e01b23544" />

Yes calibration worked for me but as I said, amplitude is also tiny for me and very flimsy.

![IMG_6696](https://github.com/user-attachments/assets/80e7ff17-3ede-4cf8-91e1-0c48b1074a6a)
![IMG_6703](https://github.com/user-attachments/assets/d3733894-fb8a-4876-8c16-528e3dd0e718)

**Comment by mattdevo1** [comment]
Thanks @segdy. I seem to have similar magnetic field strengths as you now.

What volume units are you using? CCF? And what value are you using for "Gas Meter Volume per half rotation"?

My total volume values seem to be WAY too high right now.

**Reply by rugene76** [reply]
I have the same meter with a similar issue.
I'm thinking that the magnetic field variation could be amplified by placing a strong magnet on either side of the meter. The thought is to create a field that gets disrupted by the diaphragm movement inside the meter. I just need to find some decent sized magnets.....

**Reply by eigenphase** [reply]
That’s a nice idea, worth a try! 

I’ve read somewhere that this things isn’t even magnetic so it’s changes in parasitic magnetic fields due to large metal moving that’s picked up… that causes the low magnitude. So adding a strong magnetic fields externally sounds like a promising idea. I have a pack of (smaller) neodymium magnetic. I can just try slapping them on and see if there’s any magnitude change

**Reply by eigenphase** [reply]
@rugene76 I tried it today but no luck :-( 

<img width="418" alt="image" src="https://github.com/user-attachments/assets/5734a9a0-4f27-42c0-a45e-c9dadf1dfe7f" />

The first part is the normal one, no magnet.
The one in the middle I placed the magnet on the back of the meter where I installed the sensor ... I put it maybe 10cm above the sensor. No change in Y and Z direction. X direction gets much larger but interestingly much noisier ... and the noise continues after I have turned off the gas (there should be just three half rotations as can be seen on the red trace, yet the blue continues to be noisy).

The third section is with magnet on the front of the meter (and the sensor on the back), i.e., the whole big metal part is in between. X direction becomes zero, red stays the same and one could say Y looks slightly higher strength but not much.


Did you have any luck?

**Comment by tronikos** [comment]
Do the qmc5883l_update_interval (affects ODR) and qmc5883l_oversampling (affects OSR) affect your measurements?
Try setting qmc5883l_oversampling to 512x and qmc5883l_update_interval to a big value maybe even 200ms but experiment with different values
